### ISSUE TITLE
 frcnn_roi_data_layer.cpp:220] Check failed: rois[i][DataPrepare::X2] < cols (500 vs. 500) rois[i][DataPrepare::X2] : 500

### ISSUE THREAD

**Issue by moyans** [makefile/frcnn#10]

**Comment by moyans** [comment]
I1130 15:56:00.993705  7514 solver.cpp:231] Iteration 0 (-5.23529 iter/s, 0.543508s/100 iters), loss = 5.15557  [ 0 / 12000 ] -> [ 0:0 (H:M) ]
I1130 15:56:00.993734  7514 solver.cpp:257]     Train net output #0: bbox_accuracy = 0.0078125
I1130 15:56:00.993741  7514 solver.cpp:257]     Train net output #1: loss_bbox = 0.679996 (* 2 = 1.35999 loss)
I1130 15:56:00.993746  7514 solver.cpp:257]     Train net output #2: loss_cls = 3.0499 (* 1 = 3.0499 loss)
I1130 15:56:00.993749  7514 solver.cpp:257]     Train net output #3: rpn_cls_loss = 0.719319 (* 1 = 0.719319 loss)
I1130 15:56:00.993754  7514 solver.cpp:257]     Train net output #4: rpn_loss_bbox = 0.207151 (* 1 = 0.207151 loss)
I1130 15:56:00.993760  7514 sgd_solver.cpp:148] Iteration 0, lr = 0.00033
F1130 15:56:01.850818  7526 frcnn_roi_data_layer.cpp:220] Check failed: rois[i][DataPrepare::X2] < cols (500 vs. 500) rois[i][DataPrepare::X2] : 500
*** Check failure stack trace: ***
    @     0x7fe1d750b5cd  google::LogMessage::Fail()
    @     0x7fe1d750d433  google::LogMessage::SendToLog()
    @     0x7fe1d750b15b  google::LogMessage::Flush()
    @     0x7fe1d750de1e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7fe1d7f6fcf5  caffe::Frcnn::FrcnnRoiDataLayer<>::FlipRois()
    @     0x7fe1d7f72851  caffe::Frcnn::FrcnnRoiDataLayer<>::load_batch()
    @     0x7fe1d7e55433  caffe::BasePrefetchingDataLayer<>::InternalThreadEntry()
    @     0x7fe1d7dbb0b5  caffe::InternalThread::entry()
    @     0x7fe1d7dbbf7e  boost::detail::thread_data<>::run()
    @     0x7fe1cdb8c5d5  (unknown)
    @     0x7fe1c835e6ba  start_thread
    @     0x7fe1d656f41d  clone
    @              (nil)  (unknown)
已放弃 (核心已转储)

**Comment by makefile** [comment]
You can modify your annotation file make sure that x2 is less than image width. Or change the code at `frcnn_roi_data_layer.cpp#220` from `CHECK_LT` to `CHECK_LE`. the coordinates x,y start from 0 instead of 1 (the PASCAL VOC start from 1).

**Comment by moyans** [comment]
The problem has been solved, thks.
